src/torchjd/autogram/_vjp.py

```diff
@@ -14,19 +15,45 @@
 # still support older versions of PyTorch where pytree is protected).
 
 
-def get_functional_vjp(module: nn.Module) -> Callable[[PyTree, PyTree], dict[str, Tensor]]:
+VJPType = Callable[[PyTree, PyTree], dict[str, Tensor]]
+
+
+class VJP(ABC):
     """
-    Create a VJP function for a module's forward pass with respect to its parameters. The returned
-    function takes both the input and the cotangents that can be vmaped jointly in both terms to
-    avoid providing to block diagonal jacobians.
+    Represents an abstract VJP function for a module's forward pass with respect to its parameters.
 
     :params module: The module to differentiate.
-    :returns: VJP function that takes cotangents and inputs and returns dictionary of names of
+    """
+
+    def __init__(self, module: nn.Module):
+        self.module = module
+        named_parameters = dict(module.named_parameters(recurse=False))
+        self.trainable_params = {k: v for k, v in named_parameters.items() if v.requires_grad}
+        self.frozen_params = {k: v for k, v in named_parameters.items() if not v.requires_grad}
```

Comment:
We could have only one for loop to gain epsilon time.

Reply:
IT would according to some people online. But not much, doesn't matter much unless it doesn't change readability.